command/ca/rekey.go

```diff
@@ -335,23 +325,32 @@ func rekeyCertificateAction(ctx *cli.Context) error {
 		}
 	}
 
-	var flag = false
+	var writeKey = false
 	//want to write for these cases
 	if (givenPrivate != "" && ctx.IsSet("out-key")) || (ctx.IsSet("out-key")) || (givenPrivate == "" && !(ctx.IsSet("out-key"))) {
```

Comment:
I *think* this can be summed up with 
```
if ctx .IsSet("out-key") || givenPrivate == "" {
...
}
```
Because the only way it would get to the second case is if ctx.IsSet("out-key") was not set, so we don't need to check that again. Can you check my logic here?

Reply:
I just tested that out with the different cases I was using and it works